core/src/main/java/hudson/model/TopLevelItemDescriptor.java

```diff
@@ -286,4 +286,8 @@ public static ExtensionList<TopLevelItemDescriptor> all() {
         return Items.all();
     }
 
+    @Restricted(NoExternalUse.class)
+    public static FormValidation doCheckDisplayNameOrNull(@AncestorInPath AbstractProject<?, ?> project, @QueryParameter String value) {
+        return Jenkins.get().doCheckDisplayName(value, project.getName());
```

Comment:
This cannot have been tested, because `WorkflowJob` is not an `AbstractProject`. Something like the following would be needed here (plus fix imports as needed),

```suggestion
    public static FormValidation doCheckDisplayNameOrNull(@AncestorInPath TopLevelItem<?, ?> item, @QueryParameter String value) {
        return Jenkins.get().doCheckDisplayName(value, item.getName());
```